packages/view/src/view/View.tsx

```diff
@@ -18,16 +18,25 @@ export const View = () => {
     };
   }, []);
 
-  const refreshData = async () => {
-    queryClient.invalidateQueries({ queryKey: queryKeys.data()});
-  };
-
   const { data } = useQuery({
     queryKey: queryKeys.data(),
     queryFn: () => client.data(),
     structuralSharing: false
   });
 
+  useEffect(() => {
+    const disposable = client.onNewEntry(entry => {
+      queryClient.setQueryData<RuntimeLogEntry[]>(queryKeys.data(), old => {
+        if (!old) return [entry];
+        return [...old, entry].sort((a, b) => new Date(b.timestamp).getTime() - new Date(a.timestamp).getTime());
```

Comment:
why do you need to create a date first to compare the timestamps? you should be able to also directly sort after the timestamps.